doc/KUKSA.val_data_broker/authorization.md

```diff
@@ -0,0 +1,505 @@
+  >   end-user.
+  
+  The resource owner in this context could be the OEM or the owner of a vehicle.
+
+* **Resource server**
+  > The server hosting the protected resources, capable of accepting
+  > and responding to protected resource requests using access tokens.
+
+  The resource server in this context is KUKSA.VAL.
+
+* **Client**
+  > An application making protected resource requests on behalf of the
+  > resource owner and with its authorization.  The term "client" does
+  > not imply any particular implementation characteristics (e.g.,
+  > whether the application executes on a server, a desktop, or other
+  > devices).
+
+  The client in this context can be apps distributed by the OEM or third parties.
+
+* **Authorization server**
+  > The server issuing access tokens to the client after successfully
+  > authenticating the resource owner and obtaining authorization.
+
+  The authorization server is most likely an server controlled by
+  the OEM and run by either the OEM or a third party (i.e. Bosch).
+
+The resource owner can specify the scope of access rights granted to a client by
+including a scope in the request for an access token from the authorization server.
+This access token will be given to the client and is sufficient for KUKSA.VAL to
+independently determine if the client is allowed access to a certain resource.
+
+*Note:*
+*Strictly speaking, KUKSA.VAL also needs to verify that the subject (found in `"sub"`)*
+*has the required access rights. See the RFC for more information.*
+*The working assumption, for now, is that anyone authorized to get a valid access token*
+*has the required access rights*
+
+In addition to the OAuth2 specification, The [JWT Profile for OAuth 2.0 Access Tokens](#json-web-token-jwt-profile-for-oauth-20-access-tokens)
+specifies a standardized structure for JWT access tokens. By following this standard,
+there's a greater chance for seamless integration with as many existing authorization
+solutions as possible.
+
+Finally, by using access tokens for authorization, the amount of responsibility placed
+on KUKSA.VAL, acting as a resource server, is minimized in the overall authorization
+infrastructure. This should minimize the amount of bespoke code necessary to support a
+wide range of authorization scenarios.
+
+## JWT Access Token
+
+JWT access tokens are supplied by adding an authorization header to the HTTP / gRPC requests.
+
+`Authorization: Bearer TOKEN`
+
+The TOKEN is a JWT access token as specified in the [RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068).
+
+> **JWT access token**:<br>
+    An OAuth 2.0 access token encoded in JWT format and complying with the requirements described in this specification. [[RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068#section-1.2)]
+
+The access token contains everything the server needs to verify the authenticity of the claims
+contained therein.
+
+Example of what's available in a JWT access token:
+
+Header:
+
+```yaml
+{
```

Comment:
If we do list the header, we probably should have "one-sentence",  maybe reference to spec,  what those are.
Especially wrt to alg would also be the place to say which one(s) is specifically supported by us

Reply:
Agreed